### DISCUSSION TITLE
Change server URL in meshagent.msh on multiple devices

### DISCUSSION THREAD

**Discussion by Tomtreas** [Ylianst/MeshCentral#5811] [Q&A]
Is there a way to do this as a blanket or device selection set with a single or a few commands to online devices?  I've tried to edit the .msh file with the file manager, but then when I try and restart the agent by agentupdate from the console, it seems to pull the URL from memory and doesn't accomplish my goal. If I try and use the terminal to restart the service, the shell quits as soon as the agent does and the agent never restarts.  I know i"m close, but not sure if this is possible over multiple devices or even a single device.  Thanks!

**Comment by si458** [comment]
The is no automatic way to do it. 
But the best way is as u almost do,
Change the url inside the .msh file, then restart the meshagent!
Don't use agentupdate! That will redowload the exe and redeploy the msh file from within the exe!
Instead use `service restart` which just issues the restart command to the meshagent servuce

**Reply by silversword411** [reply]
> automatic way to do it.

Sure would be nice to have some kind of way to script stuff...in the background...like Remote Management or something....maybe even add some Monitoring...wonder if anyone's thought of something like that...it would be Tactically advantageous.

**Reply by si458** [reply]
oh you must be talking about [tacticalrmm](https://github.com/amidaware/tacticalrmm) hehe

**Comment by Tomtreas** [comment]
THank you!  THe service restart allows me to do multiples after going
through and editing the URL on the msh file.  I notice something while
editing the msh files though.  Sometimes the file shows up looking like
this and the URL I'm trying to change is nowhere to be found.  SOmetimes I
can refresh the file list and it will then edit display as normal.

[image: image.png]

Normally it looks as below and I can edit save and move to the next one.

[image: image.png]

**Reply by si458** [reply]
no pictures but im guessing its a bug with the text editor in the web ui
open a github issue/bug report with all details and pics

**Reply by Tomtreas** [reply]
Yes I submitted a bug/issue.  Thanks again!
